tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+
+public struct DT 
+{
+    public Vector3 a;
+    public Vector3 b;
+};
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+public struct ComplexDT 
+{
+    public int iv;
+    public DT vecs;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst=256)] public string str;
+    public Vector3 v3;
+};
+
+class PInvokeTest 
+{
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern int nativeCall_PInvoke_CheckVector3Size();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern Vector3 nativeCall_PInvoke_Vector3Ret();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Array(Vector3[] v_array);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern DT nativeCall_PInvoke_Vector3InStruct(DT d);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_PInvoke_Vector3InComplexStruct(ref ComplexDT cdt);
+    
+    public static bool test() 
+    {
+
+        // Expected return value is 12 bytes.
+        if (nativeCall_PInvoke_CheckVector3Size() != 12) 
```

Comment:
The native passes 12 bytes.

```
  ; Vector3 arg
  000008F5: 83 EC 0C           sub         esp,0Ch 
  000008F8: 8B C4              mov         eax,esp
  000008FA: 8B 4D F0           mov         ecx,dword ptr [ebp-10h]
  000008FD: 89 08              mov         dword ptr [eax],ecx
  000008FF: 8B 55 F4           mov         edx,dword ptr [ebp-0Ch]
  00000902: 89 50 04           mov         dword ptr [eax+4],edx
  00000905: 8B 4D F8           mov         ecx,dword ptr [ebp-8]
  00000908: 89 48 08           mov         dword ptr [eax+8],ecx
  ; str arg
  0000090B: 8B 15 00 00 00 00  mov         edx,dword ptr [?str@?1??nativeCall_RPInvoke_Vector3Arg@@9@4@4PADA] 
  00000911: 52                 push        edx
  ; Vector3 arg
  00000912: 83 EC 0C           sub         esp,0Ch
  00000915: 8B C4              mov         eax,esp
  00000917: 8B 4D E4           mov         ecx,dword ptr [ebp-1Ch]
  0000091A: 89 08              mov         dword ptr [eax],ecx
  0000091C: 8B 55 E8           mov         edx,dword ptr [ebp-18h]
  0000091F: 89 50 04           mov         dword ptr [eax+4],edx
  00000922: 8B 4D EC           mov         ecx,dword ptr [ebp-14h]
  00000925: 89 48 08           mov         dword ptr [eax+8],ecx
  ; int arg
  00000928: 8B 55 E0           mov         edx,dword ptr [ebp-20h]
  0000092B: 52                 push        edx
  ; call
  0000092C: FF 55 08           call        dword ptr [ebp+8]
```

RyuJit reads only 12 bytes and the upper 4bytes are initialized to zero when loading the value to register.

```
; The address of v is stored in V00. The value of V0 is loaded to a 128-bit register and reg[127:96] is initialized to zero. So, the dot product result is correct.
Generating: N019 (  1,  1) [000015] ------------             *  lclVar    byref  V00 arg0         u:2 rcx (last use) REG rcx RV $80
Generating: N021 (  7,  5) [000018] ---XG-------             *  indir     simd12 REG mm1
							V00 in reg rcx is becoming dead  [000015]
							Live regs: 00000002 {rcx mm0} => 00000000 {mm0}
							Live vars: {V00 V01} => {V01}
							Byref regs: 00000002 {rcx} => 00000000 {}
IN000b:        vmovss   ymm2, dword ptr [rcx+8]   ; --> upper 64 bits are initialized to zero
IN000c:        vmovsd   ymm1, qword ptr [rcx]     ; --> upper 96 bits are initialized to zero
IN000d:        vshufps  ymm1, ymm2, 68            ; 68 = 44h ymm1 <= ymm2[63:32]::ymm2[31:00]::ymm1[63:32] ::ymm1[31:0]
Generating: N023 (  3,  2) [000016] ------------             *  lclVar    simd16 V01 loc0         u:3 mm0 (last use) REG mm0 RV $201
Generating: N025 ( 11,  8) [000019] ---XG-------             *  simd      float  float Dot REG mm6 $102
							V01 in reg mm0 is becoming dead  [000016]
							Live regs: 00000000 {mm0} => 00000000 {}
							Live vars: {V01} => {}
IN000e:        vmovaps  ymm6, ymm1
IN000f:        vdpps    ymm6, ymm0, 241
Generating: N027 ( 15, 13) [000021] DA-XG-------             *  st.lclVar float  V02 loc1         d:3 mm6 REG mm6 RV
```

I will try to find spec.